src/common/components/repositories-home/index.js

```diff
@@ -92,18 +96,18 @@ class RepositoriesHome extends Component {
   }
 
   render() {
-    // show spinner if snaps data was not yet fetched (snaps list is empty)
-    // (to avoid spinner during polling we are not checking `success` or `isFetching`
+    // show spinner if user has just signed in snaps data was not yet fetched
     //
-    // when snaps are loaded and user doesn't have any, they will be redirected
-    // to select repositories (so spinner won't be showing endlessly)
-    return !this.props.hasSnaps
+    // when snaps are loaded and user (who just signed in) doesn't have any,
+    // they will be redirected to select repositories
+    return (this.props.hasJustSignedIn && this.props.snaps.isFetching)
```

Comment:
The point of this change is for users to be able to see empty My repos.

If user sees empty My repos on /user/:user and refreshes they should see the same page.

The automatic redirect to select-repos is only meant to work for new users when they first sign in. Once they get back to (empty) My repos it should be OK for them to stay there after the refresh.